feat: 독서록 조회 기능 비즈니스 로직 작성

dani820 opened the pull request:
## 이 PR을 통해 해결하려는 문제
- 독서록 단 건 조회를 위한 조회 기능 개발

## 추가 및 변경사항
- 독서록 조회 기능 비즈니스 로직 작성
- 테스트 코드 작성
- 일부 도메인 엔티티 내 `@Getter` 어노테이션 추가(Content, Post, Member)

## 참고(옵션)
- 클라이언트에 조회 정보(Post) 전달 위한 PostResponse 객체 생성 위치가 적당한지에 대한 확인 필요
- uuid 항목 추가에 따른 독서록 저장 로직 변경 필요
- uuid 생성 로직 위치 정리 필요

## 체크리스트
- [x] 리뷰어를 지정하였는가
- [x] 코드가 오류나 경고없이 빌드되는가
- [x] 추가 및 변경된 사항에 대해 충분히 테스트 하였는가

dani820 committed:
feat: 독서록 조회 기능 비즈니스 로직 작성 및 테스트 코드 작성

f-lab-bradley commented on src/main/java/com/inmybook/adapter/out/InMemoryPostRepository.java:
```diff
@@ -18,4 +21,10 @@ public Long save(Post post) {
 	}
+
+	@Override
+	public Post readPostById(ReadPostCommand readPostCommand) {
```
1. findPostById 가 나을 것 같아요
2. Command 는 application layer 에서 사용하는건데 domain layer 까지 끌고 들어왔네요. byId 니까 id 만 파라미터로 받는게 좋겠어요

dani820 replied:
수정하겠습니다~

f-lab-bradley commented on src/main/java/com/inmybook/domain/member/Member.java:
```diff
@@ -1,14 +1,12 @@
 @Builder
 public class Member {
-	private String memberNo;
+	private Long memberNo;
```
왜 long 을 사용 안하고 Long 을 사용하셨을까요???

dani820 replied:
auto increment로 pk 값을 생성한다고 가정하고 작성했습니다. 데이터 등록 시 해당 pk 값은 db 쪽에서 생성될 것이기 때문에 Member 객체 생성 시점에 pk 값을 null 로 초기화하기 위해서 Long을 사용했습니다

f-lab-bradley replied:
member 는 도메인이죠????
도메인이 pk 값을 알 필요가 있을까요???
만약  auto increment 를 지원하지 않는 db 라면 해당 값은 도메인이 어떻게 처리해야 하는거죠???
pk 값은 db 쪽과 직접 연결된 곳만 알면 되지 않을까요???

😄 (화난게 아니라는 표시)

dani820 replied:
이해했습니다! 수정하겠습니다~

f-lab-bradley commented on src/main/java/com/inmybook/application/service/BookResponse.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Builder;
+
+@Builder
+public record BookResponse(
```
Response 로 네이밍 한 이유는 무엇인가요??

dani820 replied:
응답 객체로 내려줄 생각을 하다보니 Response로 네이밍하게 되었습니다,,

f-lab-bradley replied:
이건 application 인데 Http 응답을 알아야 할 필요가 있을까요???

dani820 replied:
아니요..! 수정하겠습니다~

f-lab-bradley commented on src/main/java/com/inmybook/application/service/ReadPostService.java:
```diff
@@ -0,0 +1,46 @@
+		Post post = readPostPort.readPostById(readPostCommand);
+
+		ContentResponse contentResponse = ContentResponse.builder()
+			.title(post.getContent().getTitle())
```
기차 충돌 문제에 대해서 알아보시겠어요?

dani820 replied:
넵 확인하겠습니다

f-lab-bradley commented:
Approve 안하면 merge 안되도록 github 세팅 해 주세요~~

dani820 commented:
변경하겠습니다~

f-lab-bradley commented on src/test/java/com/inmybook/application/service/ReadPostServiceTest.java:
```diff
@@ -61,7 +61,7 @@ void readPost() {
 		ReadPostCommand mockReadPostCommand = new ReadPostCommand(bookUuid);
 
-		when(readPostPort.findPostById(mockReadPostCommand)).thenReturn(post);
+		when(readPostPort.findPostById(mockReadPostCommand.uuid())).thenReturn(post);
```
uuid 라는 표현은 java 에서 id 생성 방식 인거고
해당 필드의 이름은 ~~Id 가 낫지 않을까 싶은데 어떻게 생각하시나요?

dani820 replied:
그렇지 않아도 네이밍이 적절한지 내심 고민하고 있던 지점이었는데요~ 수정하겠습니다!

f-lab-bradley commented on src/main/java/com/inmybook/adapter/out/InMemoryPostRepository.java:
```diff
@@ -4,11 +4,12 @@
 
 @Repository
-public class InMemoryPostRepository implements RegisterPostPort {
+public class InMemoryPostRepository implements RegisterPostPort, ReadPostPort {
```
Mixin pattern 에 대해서도 학습하면 좋을 것 같아요 ㅎㅎ

dani820 replied:
알아보겠습니다~

dani820 committed:
refactor: 서비스 레이어에서 요청, 응답 시 사용하는 dto 클래스명 변경

요청 dto : ~Command -> ~Input
응답 dto : ~Response -> ~DetailsOuput 으로 변경

dani820 committed:
refactor: 도메인 엔티티 내 id 값 변경

각 도메인 엔티티 내에서 auto increment id 사용 부분 변경 혹은 삭제
Thumbnail 엔티티 내 생성자 제거 -> 어노테이션 사용

dani820 committed:
refactor: id 변수명 변경 (uuid -> postId)

f-lab-bradley reviewed (approved):
DTO 관련 테스트 코드가 하나도 없어요.
이거 테스트 보완 이슈 추가 해 주세요
